sla_per_use_system_test.py

```diff
@@ -0,0 +1,394 @@
+                                                      'round_robin': True})
+
+        for queue in read_queue:
+            self.verify_stress_thread(queue=queue)
+
+        results = {}
+        for i, read in enumerate(read_queue):
+            res = self.get_stress_results(queue=read, store_results=False)
+            op_rate, username = None, None
+            if res:
+                op_rate = res[0].get('op rate')
+                username = res[0].get('username')
+
+            if not (op_rate and username):
+                self.log.error('Stress statistics are not received for user {}. Can\'t complete the test'
+                               .format(users_names[i]))
+                return None
+            self.assertEqual(username, users_names[i], msg='Expected that stress was run with user "{}" but it was "{}"'
+                             .format(users_names[i], username))
+            results[username] = int(op_rate)
+
+        end = int(time.time())
+        scylla_load = self.prometheus_stats.get_scylla_reactor_utilization(start_time=start, end_time=end)
+        if scylla_load < waited_cpu_min:
+            self.log.warning('Load isn\'t high enough. The test results may be not correct')
+
+        return results
+
+    def clean_auth(self, entities_list_of_dict):
+        for entity in entities_list_of_dict:
+            sl = entity.get('service_level')
+            role = entity.get('role')
+            user = entity.get('user')
+            if user:
+                user.drop()
+            if role:
+                role.drop()
+            if sl:
+                sl.drop()
+
+        self.backgroud_task = None
+
+    # TODO: decide if need that
+    # def create_background_load(self, duration='120m', timeout=900, cpu_minimum=99, sleep_time_sec=120,
+    #                            rows_increase=500000):
+    #     stress_cmd = self.STRESS_WRITE_DURATION_CMD.format(duration=duration, user=self.background_user[0],
+    #                                                        password=self.background_user[0],
+    #                                                        threads=250, pop='%d..%d' % (self.n, self.n+rows_increase))
+    #     self.backgroud_task = self.run_stress_thread(stress_cmd=stress_cmd)
+    #
+    #     return wait.wait_for(func=self.wait_for_load_reached, step=sleep_time_sec, timeout=timeout,
+    #                          cpu_minimum=cpu_minimum, sleep_time_sec=sleep_time_sec)
+    #
+    # def wait_for_load_reached(self, cpu_minimum=99, sleep_time_sec=120):
+    #     end = int(time.time())
+    #     start = int(time.time())-sleep_time_sec
+    #     cpu = self.prometheus_stats.get_scylla_reactor_utilization(start_time=start, end_time=end)
+    #     if cpu and cpu > cpu_minimum:
+    #         self.log.debug("CPU load reached {}%".format(cpu_minimum))
+    #         return cpu > cpu_minimum
+    #     return False
+
+    def test_run_all(self):
```

Comment:
I'm not sure this should be the approach to run all tests.
Let's discuss it with @fruch and @bentsi.
The question is more general than this case, how do we run a set of tests in a module post avocado era?
Is it something the infra should take care of?
Can we "reuse" cluster between tests?

Reply:
changed